openedx/core/djangoapps/programs/tests/test_utils.py

```diff
@@ -967,6 +972,65 @@ def test_multiple_published_course_runs(self):
 
         self.assertTrue(data['is_learner_eligible_for_one_click_purchase'])
 
+    def test_learner_eligibility_for_one_click_purchase_entitlement_products(self):
+        """
+        Learner should be eligible for one click purchase if:
+            - program is eligible for one click purchase
+            - There are courses remaining that have not been purchased
+        """
+        course1 = _create_course(self, self.course_price, course_run_count=2, entitlement_count=1)
+        course2 = _create_course(self, self.course_price, course_run_count=2, entitlement_count=1)
+        expected_skus = set([course1['entitlements'][0]['sku'], course2['entitlements'][0]['sku']])
+        program = ProgramFactory(
+            courses=[course1, course2],
+            is_program_eligible_for_one_click_purchase=True,
+            applicable_seat_types=['verified'],
+        )
+        data = ProgramDataExtender(program, self.user).extend()
+        self.assertTrue(data['is_learner_eligible_for_one_click_purchase'])
+        self.assertEqual(set(data['skus']), expected_skus)
```

Comment:
Can we add some assertions (sanity checks) that verify that this is `False` when either of the two conditions mentioned in the docstring are `False`?

Reply:
Nevermind, it looks like you added some of them below. It might be nice to have a test for the case where there are entitlements that haven't been purchased but `is_program_eligible_for_one_click_purchase` has been set to `False` for the program though.